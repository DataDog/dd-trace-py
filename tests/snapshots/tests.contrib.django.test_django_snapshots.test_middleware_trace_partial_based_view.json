[[
  {
    "name": "django.request",
    "service": "django",
    "resource": "GET ^partial-view/$",
    "trace_id": 0,
    "span_id": 1,
    "parent_id": 0,
    "type": "web",
    "error": 0,
    "meta": {
      "_dd.p.dm": "-0",
      "component": "django",
      "django.request.class": "django.core.handlers.wsgi.WSGIRequest",
      "django.response.class": "django.http.response.HttpResponse",
      "django.user.is_authenticated": "False",
      "django.view": "partial-view",
      "http.method": "GET",
      "http.route": "^partial-view/$",
      "http.status_code": "200",
      "http.url": "http://testserver/partial-view/",
      "language": "python",
      "runtime-id": "99c984f231264c238e23d78a016b9ddd",
      "span.kind": "server"
    },
    "metrics": {
      "_dd.agent_psr": 1.0,
      "_dd.measured": 1,
      "_dd.top_level": 1,
      "_dd.tracer_kr": 1.0,
      "_sampling_priority_v1": 1,
      "process_id": 11874
    },
    "duration": 2009910,
    "start": 1691162188898676679
  },
     {
       "name": "django.middleware",
       "service": "django",
       "resource": "django.contrib.sessions.middleware.SessionMiddleware.__call__",
       "trace_id": 0,
       "span_id": 2,
       "parent_id": 1,
       "type": "",
       "error": 0,
       "meta": {
         "_dd.p.dm": "-0",
         "component": "django"
       },
       "duration": 1647394,
       "start": 1691162188898817943
     },
        {
          "name": "django.middleware",
          "service": "django",
          "resource": "django.contrib.sessions.middleware.SessionMiddleware.process_request",
          "trace_id": 0,
          "span_id": 3,
          "parent_id": 2,
          "type": "",
          "error": 0,
          "meta": {
            "_dd.p.dm": "-0",
            "component": "django"
          },
          "duration": 47839,
          "start": 1691162188898871541
        },
        {
          "name": "django.middleware",
          "service": "django",
          "resource": "django.middleware.common.CommonMiddleware.__call__",
          "trace_id": 0,
          "span_id": 4,
          "parent_id": 2,
          "type": "",
          "error": 0,
          "meta": {
            "_dd.p.dm": "-0",
            "component": "django"
          },
          "duration": 1453536,
          "start": 1691162188898952691
        },
           {
             "name": "django.middleware",
             "service": "django",
             "resource": "django.middleware.common.CommonMiddleware.process_request",
             "trace_id": 0,
             "span_id": 6,
             "parent_id": 4,
             "type": "",
             "error": 0,
             "meta": {
               "_dd.p.dm": "-0",
               "component": "django"
             },
             "duration": 50105,
             "start": 1691162188898999228
           },
           {
             "name": "django.middleware",
             "service": "django",
             "resource": "django.middleware.csrf.CsrfViewMiddleware.__call__",
             "trace_id": 0,
             "span_id": 7,
             "parent_id": 4,
             "type": "",
             "error": 0,
             "meta": {
               "_dd.p.dm": "-0",
               "component": "django"
             },
             "duration": 1268792,
             "start": 1691162188899074821
           },
              {
                "name": "django.middleware",
                "service": "django",
                "resource": "django.middleware.csrf.CsrfViewMiddleware.process_request",
                "trace_id": 0,
                "span_id": 9,
                "parent_id": 7,
                "type": "",
                "error": 0,
                "meta": {
                  "_dd.p.dm": "-0",
                  "component": "django"
                },
                "duration": 25497,
                "start": 1691162188899112081
              },
              {
                "name": "django.middleware",
                "service": "django",
                "resource": "django.contrib.auth.middleware.AuthenticationMiddleware.__call__",
                "trace_id": 0,
                "span_id": 10,
                "parent_id": 7,
                "type": "",
                "error": 0,
                "meta": {
                  "_dd.p.dm": "-0",
                  "component": "django"
                },
                "duration": 1130849,
                "start": 1691162188899160362
              },
                 {
                   "name": "django.middleware",
                   "service": "django",
                   "resource": "django.contrib.auth.middleware.AuthenticationMiddleware.process_request",
                   "trace_id": 0,
                   "span_id": 12,
                   "parent_id": 10,
                   "type": "",
                   "error": 0,
                   "meta": {
                     "_dd.p.dm": "-0",
                     "component": "django"
                   },
                   "duration": 29976,
                   "start": 1691162188899203445
                 },
                 {
                   "name": "django.middleware",
                   "service": "django",
                   "resource": "django.contrib.messages.middleware.MessageMiddleware.__call__",
                   "trace_id": 0,
                   "span_id": 13,
                   "parent_id": 10,
                   "type": "",
                   "error": 0,
                   "meta": {
                     "_dd.p.dm": "-0",
                     "component": "django"
                   },
                   "duration": 1023784,
                   "start": 1691162188899256923
                 },
                    {
                      "name": "django.middleware",
                      "service": "django",
                      "resource": "django.contrib.messages.middleware.MessageMiddleware.process_request",
                      "trace_id": 0,
                      "span_id": 14,
                      "parent_id": 13,
                      "type": "",
                      "error": 0,
                      "meta": {
                        "_dd.p.dm": "-0",
                        "component": "django"
                      },
                      "duration": 62094,
                      "start": 1691162188899292064
                    },
                    {
                      "name": "django.middleware",
                      "service": "django",
                      "resource": "django.middleware.clickjacking.XFrameOptionsMiddleware.__call__",
                      "trace_id": 0,
                      "span_id": 15,
                      "parent_id": 13,
                      "type": "",
                      "error": 0,
                      "meta": {
                        "_dd.p.dm": "-0",
                        "component": "django"
                      },
                      "duration": 846522,
                      "start": 1691162188899377750
                    },
                       {
                         "name": "django.middleware",
                         "service": "django",
                         "resource": "django.middleware.security.SecurityMiddleware.__call__",
                         "trace_id": 0,
                         "span_id": 17,
                         "parent_id": 15,
                         "type": "",
                         "error": 0,
                         "meta": {
                           "_dd.p.dm": "-0",
                           "component": "django"
                         },
                         "duration": 743651,
                         "start": 1691162188899414029
                       },
                          {
                            "name": "django.middleware",
                            "service": "django",
                            "resource": "django.middleware.security.SecurityMiddleware.process_request",
                            "trace_id": 0,
                            "span_id": 19,
                            "parent_id": 17,
                            "type": "",
                            "error": 0,
                            "meta": {
                              "_dd.p.dm": "-0",
                              "component": "django"
                            },
                            "duration": 22122,
                            "start": 1691162188899448113
                          },
                          {
                            "name": "django.middleware",
                            "service": "django",
                            "resource": "tests.contrib.django.middleware.ClsMiddleware.__call__",
                            "trace_id": 0,
                            "span_id": 20,
                            "parent_id": 17,
                            "type": "",
                            "error": 0,
                            "meta": {
                              "_dd.p.dm": "-0",
                              "component": "django"
                            },
                            "duration": 591993,
                            "start": 1691162188899492226
                          },
                             {
                               "name": "django.middleware",
                               "service": "django",
                               "resource": "tests.contrib.django.middleware.fn_middleware",
                               "trace_id": 0,
                               "span_id": 22,
                               "parent_id": 20,
                               "type": "",
                               "error": 0,
                               "meta": {
                                 "_dd.p.dm": "-0",
                                 "component": "django"
                               },
                               "duration": 548771,
                               "start": 1691162188899525219
                             },
                                {
                                  "name": "django.middleware",
                                  "service": "django",
                                  "resource": "tests.contrib.django.middleware.EverythingMiddleware.__call__",
                                  "trace_id": 0,
                                  "span_id": 23,
                                  "parent_id": 22,
                                  "type": "",
                                  "error": 0,
                                  "meta": {
                                    "_dd.p.dm": "-0",
                                    "component": "django"
                                  },
                                  "duration": 496843,
                                  "start": 1691162188899566189
                                },
                                   {
                                     "name": "django.middleware",
                                     "service": "django",
                                     "resource": "django.middleware.csrf.CsrfViewMiddleware.process_view",
                                     "trace_id": 0,
                                     "span_id": 24,
                                     "parent_id": 23,
                                     "type": "",
                                     "error": 0,
                                     "meta": {
                                       "_dd.p.dm": "-0",
                                       "component": "django"
                                     },
                                     "duration": 32609,
                                     "start": 1691162188899869294
                                   },
                                   {
                                     "name": "django.middleware",
                                     "service": "django",
                                     "resource": "tests.contrib.django.middleware.EverythingMiddleware.process_view",
                                     "trace_id": 0,
                                     "span_id": 25,
                                     "parent_id": 23,
                                     "type": "",
                                     "error": 0,
                                     "meta": {
                                       "_dd.p.dm": "-0",
                                       "component": "django"
                                     },
                                     "duration": 22216,
                                     "start": 1691162188899926481
                                   },
                                   {
                                     "name": "django.view",
                                     "service": "django",
                                     "resource": "partial",
                                     "trace_id": 0,
                                     "span_id": 26,
                                     "parent_id": 23,
                                     "type": "",
                                     "error": 0,
                                     "meta": {
                                       "_dd.p.dm": "-0",
                                       "component": "django"
                                     },
                                     "duration": 50840,
                                     "start": 1691162188899993838
                                   },
                          {
                            "name": "django.middleware",
                            "service": "django",
                            "resource": "django.middleware.security.SecurityMiddleware.process_response",
                            "trace_id": 0,
                            "span_id": 21,
                            "parent_id": 17,
                            "type": "",
                            "error": 0,
                            "meta": {
                              "_dd.p.dm": "-0",
                              "component": "django"
                            },
                            "duration": 38073,
                            "start": 1691162188900107855
                          },
                       {
                         "name": "django.middleware",
                         "service": "django",
                         "resource": "django.middleware.clickjacking.XFrameOptionsMiddleware.process_response",
                         "trace_id": 0,
                         "span_id": 18,
                         "parent_id": 15,
                         "type": "",
                         "error": 0,
                         "meta": {
                           "_dd.p.dm": "-0",
                           "component": "django"
                         },
                         "duration": 33678,
                         "start": 1691162188900179541
                       },
                    {
                      "name": "django.middleware",
                      "service": "django",
                      "resource": "django.contrib.messages.middleware.MessageMiddleware.process_response",
                      "trace_id": 0,
                      "span_id": 16,
                      "parent_id": 13,
                      "type": "",
                      "error": 0,
                      "meta": {
                        "_dd.p.dm": "-0",
                        "component": "django"
                      },
                      "duration": 24520,
                      "start": 1691162188900245773
                    },
              {
                "name": "django.middleware",
                "service": "django",
                "resource": "django.middleware.csrf.CsrfViewMiddleware.process_response",
                "trace_id": 0,
                "span_id": 11,
                "parent_id": 7,
                "type": "",
                "error": 0,
                "meta": {
                  "_dd.p.dm": "-0",
                  "component": "django"
                },
                "duration": 21408,
                "start": 1691162188900312054
              },
           {
             "name": "django.middleware",
             "service": "django",
             "resource": "django.middleware.common.CommonMiddleware.process_response",
             "trace_id": 0,
             "span_id": 8,
             "parent_id": 4,
             "type": "",
             "error": 0,
             "meta": {
               "_dd.p.dm": "-0",
               "component": "django"
             },
             "duration": 29139,
             "start": 1691162188900366125
           },
        {
          "name": "django.middleware",
          "service": "django",
          "resource": "django.contrib.sessions.middleware.SessionMiddleware.process_response",
          "trace_id": 0,
          "span_id": 5,
          "parent_id": 2,
          "type": "",
          "error": 0,
          "meta": {
            "_dd.p.dm": "-0",
            "component": "django"
          },
          "duration": 27839,
          "start": 1691162188900427175
        }]]
