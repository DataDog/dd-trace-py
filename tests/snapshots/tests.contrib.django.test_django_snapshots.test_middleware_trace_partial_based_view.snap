[[{"name" "django.request"
   "service" "django"
   "resource" "GET ^partial-view/$"
   "type" "web"
   "error" 0
   "span_id" 0
   "trace_id" 0
   "parent_id" nil
   "start" 1612561069947281000
   "duration" 1942000
   "meta" {"runtime-id" "ed399477c62a49f8a3205405cacc59c9"
           "http.route" "^partial-view/$"
           "http.url" "http://testserver/partial-view/"
           "django.user.is_authenticated" "False"
           "http.status_code" "200"
           "http.method" "GET"
           "django.response.class" "django.http.response.HttpResponse"
           "django.request.class" "django.core.handlers.wsgi.WSGIRequest"
           "django.view" "partial-view"}
   "metrics" {"_dd.agent_psr" 1.0
              "_dd.measured" 1
              "_sampling_priority_v1" 1
              "_dd.tracer_kr" 1.0
              "system.pid" 97933}}
      {"name" "django.middleware"
       "service" "django"
       "resource" "django.contrib.sessions.middleware.SessionMiddleware.__call__"
       "error" 0
       "span_id" 1
       "trace_id" 0
       "parent_id" 0
       "start" 1612561069947368000
       "duration" 1673000}
          {"name" "django.middleware"
           "service" "django"
           "resource" "django.contrib.sessions.middleware.SessionMiddleware.process_request"
           "error" 0
           "span_id" 2
           "trace_id" 0
           "parent_id" 1
           "start" 1612561069947393000
           "duration" 55000}
          {"name" "django.middleware"
           "service" "django"
           "resource" "django.middleware.common.CommonMiddleware.__call__"
           "error" 0
           "span_id" 3
           "trace_id" 0
           "parent_id" 1
           "start" 1612561069947473000
           "duration" 1531000}
              {"name" "django.middleware"
               "service" "django"
               "resource" "django.middleware.common.CommonMiddleware.process_request"
               "error" 0
               "span_id" 5
               "trace_id" 0
               "parent_id" 3
               "start" 1612561069947496000
               "duration" 42000}
              {"name" "django.middleware"
               "service" "django"
               "resource" "django.middleware.csrf.CsrfViewMiddleware.__call__"
               "error" 0
               "span_id" 6
               "trace_id" 0
               "parent_id" 3
               "start" 1612561069947562000
               "duration" 1402000}
                  {"name" "django.middleware"
                   "service" "django"
                   "resource" "django.middleware.csrf.CsrfViewMiddleware.process_request"
                   "error" 0
                   "span_id" 8
                   "trace_id" 0
                   "parent_id" 6
                   "start" 1612561069947585000
                   "duration" 11000}
                  {"name" "django.middleware"
                   "service" "django"
                   "resource" "django.contrib.auth.middleware.AuthenticationMiddleware.__call__"
                   "error" 0
                   "span_id" 9
                   "trace_id" 0
                   "parent_id" 6
                   "start" 1612561069947614000
                   "duration" 1318000}
                      {"name" "django.middleware"
                       "service" "django"
                       "resource" "django.contrib.auth.middleware.AuthenticationMiddleware.process_request"
                       "error" 0
                       "span_id" 11
                       "trace_id" 0
                       "parent_id" 9
                       "start" 1612561069947635000
                       "duration" 14000}
                      {"name" "django.middleware"
                       "service" "django"
                       "resource" "django.contrib.messages.middleware.MessageMiddleware.__call__"
                       "error" 0
                       "span_id" 12
                       "trace_id" 0
                       "parent_id" 9
                       "start" 1612561069947667000
                       "duration" 1260000}
                          {"name" "django.middleware"
                           "service" "django"
                           "resource" "django.contrib.messages.middleware.MessageMiddleware.process_request"
                           "error" 0
                           "span_id" 13
                           "trace_id" 0
                           "parent_id" 12
                           "start" 1612561069947839000
                           "duration" 53000}
                          {"name" "django.middleware"
                           "service" "django"
                           "resource" "django.middleware.clickjacking.XFrameOptionsMiddleware.__call__"
                           "error" 0
                           "span_id" 14
                           "trace_id" 0
                           "parent_id" 12
                           "start" 1612561069947913000
                           "duration" 980000}
                              {"name" "django.middleware"
                               "service" "django"
                               "resource" "django.middleware.security.SecurityMiddleware.__call__"
                               "error" 0
                               "span_id" 16
                               "trace_id" 0
                               "parent_id" 14
                               "start" 1612561069947936000
                               "duration" 915000}
                                  {"name" "django.middleware"
                                   "service" "django"
                                   "resource" "django.middleware.security.SecurityMiddleware.process_request"
                                   "error" 0
                                   "span_id" 18
                                   "trace_id" 0
                                   "parent_id" 16
                                   "start" 1612561069947959000
                                   "duration" 9000}
                                  {"name" "django.middleware"
                                   "service" "django"
                                   "resource" "tests.contrib.django.middleware.ClsMiddleware.__call__"
                                   "error" 0
                                   "span_id" 19
                                   "trace_id" 0
                                   "parent_id" 16
                                   "start" 1612561069947986000
                                   "duration" 826000}
                                      {"name" "django.middleware"
                                       "service" "django"
                                       "resource" "tests.contrib.django.middleware.EverythingMiddleware"
                                       "error" 0
                                       "span_id" 21
                                       "trace_id" 0
                                       "parent_id" 19
                                       "start" 1612561069948007000
                                       "duration" 801000}
                                          {"name" "django.middleware"
                                           "service" "django"
                                           "resource" "tests.contrib.django.middleware.EverythingMiddleware.__call__"
                                           "error" 0
                                           "span_id" 22
                                           "trace_id" 0
                                           "parent_id" 21
                                           "start" 1612561069948027000
                                           "duration" 775000}
                                              {"name" "django.middleware"
                                               "service" "django"
                                               "resource" "django.middleware.csrf.CsrfViewMiddleware.process_view"
                                               "error" 0
                                               "span_id" 23
                                               "trace_id" 0
                                               "parent_id" 22
                                               "start" 1612561069948101000
                                               "duration" 13000}
                                              {"name" "django.middleware"
                                               "service" "django"
                                               "resource" "tests.contrib.django.middleware.EverythingMiddleware.process_view"
                                               "error" 0
                                               "span_id" 24
                                               "trace_id" 0
                                               "parent_id" 22
                                               "start" 1612561069948131000
                                               "duration" 9000}
                                              {"name" "django.view"
                                               "service" "django"
                                               "resource" "partial"
                                               "error" 0
                                               "span_id" 25
                                               "trace_id" 0
                                               "parent_id" 22
                                               "start" 1612561069948169000
                                               "duration" 625000}
                                  {"name" "django.middleware"
                                   "service" "django"
                                   "resource" "django.middleware.security.SecurityMiddleware.process_response"
                                   "error" 0
                                   "span_id" 20
                                   "trace_id" 0
                                   "parent_id" 16
                                   "start" 1612561069948836000
                                   "duration" 11000}
                              {"name" "django.middleware"
                               "service" "django"
                               "resource" "django.middleware.clickjacking.XFrameOptionsMiddleware.process_response"
                               "error" 0
                               "span_id" 17
                               "trace_id" 0
                               "parent_id" 14
                               "start" 1612561069948871000
                               "duration" 15000}
                          {"name" "django.middleware"
                           "service" "django"
                           "resource" "django.contrib.messages.middleware.MessageMiddleware.process_response"
                           "error" 0
                           "span_id" 15
                           "trace_id" 0
                           "parent_id" 12
                           "start" 1612561069948912000
                           "duration" 11000}
                  {"name" "django.middleware"
                   "service" "django"
                   "resource" "django.middleware.csrf.CsrfViewMiddleware.process_response"
                   "error" 0
                   "span_id" 10
                   "trace_id" 0
                   "parent_id" 6
                   "start" 1612561069948950000
                   "duration" 10000}
              {"name" "django.middleware"
               "service" "django"
               "resource" "django.middleware.common.CommonMiddleware.process_response"
               "error" 0
               "span_id" 7
               "trace_id" 0
               "parent_id" 3
               "start" 1612561069948982000
               "duration" 17000}
          {"name" "django.middleware"
           "service" "django"
           "resource" "django.contrib.sessions.middleware.SessionMiddleware.process_response"
           "error" 0
           "span_id" 4
           "trace_id" 0
           "parent_id" 1
           "start" 1612561069949022000
           "duration" 15000}]]
